graph/src/components/ethereum/types.rs

```diff
@@ -81,32 +81,59 @@ pub struct EthereumCall {
     pub transaction_hash: Option<H256>,
 }
 
-impl From<&Trace> for EthereumCall {
-    fn from(trace: &Trace) -> Self {
-        let (from, to, value, input) = match &trace.action {
-            Action::Call(call) => (call.from, call.to, call.value, call.input.clone()),
-            _ => (
-                Address::zero(),
-                Address::zero(),
-                U256::zero(),
-                Bytes::from(vec![]),
-            ),
+impl EthereumCall {
+    pub fn try_from_trace(trace: &Trace) -> Option<Self> {
+        if trace.result.is_none() || trace.error.is_some() {
```

Comment:
`trace.result.is_none()` is already checked in the `match` below. In what cases do we get errors? Should we justify why the error content is discarded?